pkg/cache/cache.go

```diff
@@ -0,0 +1,130 @@
+package cache
+
+import (
+	"time"
```

Comment:
The test for this package can be made less racy by using a mockable clock such as https://github.com/benbjohnson/clock embedded in the cache. It satisfies the interface of the functions in the standard library time package (meaning supports Ticker and such).